### ISSUE TITLE
Cannot connect to kong

### ISSUE THREAD

**Issue by githandyman** [pantsel/konga#195]
Receive an error message with trying to activate connection:  "Oh snap! Can't connect to http://kong:8001"
![image](https://user-images.githubusercontent.com/29261482/37714723-3fa6cf28-2ce8-11e8-8456-4d732803b627.png)

Using Rancher Server to administrate docker cluster using Cattle orchestration.  Created a Rancher Stack using the docker compose file.   There is no issue with the Konga container launching or connecting to Postgres database.  However, Konga will not connect to Kong.  I am able to ping and netcat (echo "GET /" | nc kong 8001) from the Konga container to kong with success and returning response from the kong admin API. 

Furthermore, I am able to successfully launch and access kong using kong-dashboard in the same Stack services group.  It's does not appear to be a networking issue.   I am able to add/delete/modify a connection in Konga; so it is not a database issue either.

I've tried launching different versions but I get the same error.

**Comment by pantsel** [comment]
@githandyman , can you post the docker compose file?

**Comment by githandyman** [comment]
![image](https://user-images.githubusercontent.com/29261482/37717956-9f8bd8dc-2cef-11e8-8360-0c7d999b5111.png)

**Comment by githandyman** [comment]
I need a little direction...Kong server logs or Chrome developer tools server logs?

**Comment by pantsel** [comment]
Konga's container logs.

**Comment by githandyman** [comment]
3/21/2018 10:12:02 AM  username: 'admin',
3/21/2018 10:12:02 AM  email: 'admin@some.domain',
3/21/2018 10:12:02 AM  firstName: 'Arnold',
3/21/2018 10:12:02 AM  lastName: 'Administrator',
3/21/2018 10:12:02 AM  admin: true,
3/21/2018 10:12:02 AM  node_id: 'http://kong:8001',
3/21/2018 10:12:02 AM  active: true,
3/21/2018 10:12:02 AM  createdAt: '2018-03-21T02:21:02.682Z',
3/21/2018 10:12:02 AM  updatedAt: '2018-03-21T02:21:02.682Z' }
3/21/2018 10:12:02 AMdebug: Passport:Policy:local:findUserPassport:error => null
3/21/2018 10:12:02 AMdebug: Passport:Policy:local:findUserPassport => { user: 1,
3/21/2018 10:12:02 AM  id: 1,
3/21/2018 10:12:02 AM  protocol: 'local',
3/21/2018 10:12:02 AM  password: '$2a$10$1UuWJ/VjhRwCNDOWpN6dheMYhQCmcSu4D/6HWHJ0NNPzy9BLRL5Je',
3/21/2018 10:12:02 AM  createdAt: '2018-03-21T02:21:02.971Z',
3/21/2018 10:12:02 AM  updatedAt: '2018-03-21T02:21:02.971Z' }
3/21/2018 10:12:47 AMdebug: Policy:dynamicNode `connection-id` is defined.
3/21/2018 10:12:47 AMdebug: KongProxyController:req.method GET
3/21/2018 10:12:47 AMdebug: KongProxyController:req.url ?connection_id=1
3/21/2018 10:12:48 AMerror: KongProxyController request error { message: 'Cannot parse JSON body' }

**Comment by pantsel** [comment]
what version of Kong & Konga are you using?

**Comment by githandyman** [comment]
kong version: 0.8.3
KONGA 0.10.6   
kong image:    littlebaydigital/kong:0.8.3
Konga image:   pantsel/konga:latest

**Comment by pantsel** [comment]
There's no way Konga will work with that version of Kong. 
It is ancient, even before Konga was born :).

Kong is now at 0.12.2 and 0.13.x is on it's way.

Check Konga's README for compatibility.
https://github.com/pantsel/konga#compatibility

I would suggest you upgrade Kong to a newer version.

**Comment by githandyman** [comment]
Can you provide a solid docker-compose file for Kong and Konga?

**Comment by pantsel** [comment]
```
version: '2'
volumes:
  db-data: {}
services:
  konga:
    image: pantsel/konga:latest
    environment:
      NODE_ENV: production
    stdin_open: true
    volumes:
    - /var/data/konga:/app/kongadata
    tty: true
    links:
    - kong:kong
    ports:
    - 1337:1337/tcp
  postgres:
    image: postgres:9.5
    environment:
      POSTGRES_DB: kong
      POSTGRES_USER: kong
    volumes:
    - db-data:/var/lib/postgresql/data
  #Setup kong database
  kong-migration:
    image: kong:latest
    environment:
      - KONG_DATABASE=postgres
      - KONG_PG_HOST=kong-database
    command: kong migrations up
  kong:
    image: kong:latest
    environment:
      KONG_DATABASE: postgres
      KONG_PG_HOST: kong-database
      KONG_ADMIN_LISTEN: 0.0.0.0:8001
      KONG_ADMIN_LISTEN_SSL: 0.0.0.0:8444
    links:
    - postgres:kong-database
    ports:
    - 8000:8000/tcp
    - 8001:8001/tcp
    - 8443:8443/tcp
    - 8444:8444/tcp


```

**Comment by nohorbee** [comment]
Hi!
I'm having the same issues but I've checked the thread and my config seems to be OK.
I'm using both images (Kong and Konga) latest which brought:
- Konga: 0.14.3
- Kong: 1.1.2

I'm not using the recommended docker-compose.yml since it's failing for other reasons. Instead, I'm using the docker-composer.yaml recommended by Kong with the Konga addition, but I'm pretty sure that's not where the problem is. I'm pasting it here anyway:
```
version: '2.1'

volumes:
    kong_data: {}
    mongo_data:

networks:
  kong-net:
services:
  kong-migrations:
    image: "${KONG_DOCKER_TAG:-kong:latest}"
    command: kong migrations bootstrap
    depends_on:
      db:
        condition: service_healthy
    environment:
      KONG_DATABASE: postgres
      KONG_PG_DATABASE: ${KONG_PG_DATABASE:-kong}
      KONG_PG_HOST: db
      KONG_PG_PASSWORD: ${KONG_PG_PASSWORD:-kong}
      KONG_PG_USER: ${KONG_PG_USER:-kong}
    links:
      - db:db
    networks:
      - kong-net
    restart: on-failure
  kong:
    image: "${KONG_DOCKER_TAG:-kong:latest}"
    user: "${KONG_USER:-root}"
    depends_on:
      db:
        condition: service_healthy
    environment:
      KONG_ADMIN_ACCESS_LOG: /dev/stdout
      KONG_ADMIN_ERROR_LOG: /dev/stderr
      KONG_ADMIN_LISTEN: '0.0.0.0:8001'
      KONG_CASSANDRA_CONTACT_POINTS: db
      KONG_DATABASE: postgres
      KONG_PG_DATABASE: ${KONG_PG_DATABASE:-kong}
      KONG_PG_HOST: db
      KONG_PG_PASSWORD: ${KONG_PG_PASSWORD:-kong}
      KONG_PG_USER: ${KONG_PG_USER:-kong}
      KONG_PROXY_ACCESS_LOG: /dev/stdout
      KONG_PROXY_ERROR_LOG: /dev/stderr
    networks:
      - kong-net
    ports:
      - "8000:8000/tcp"
      - "8001:8001/tcp"
      - "8443:8443/tcp"
      - "8444:8444/tcp"
    restart: on-failure
  db:
    image: postgres:9.5
    environment:
      POSTGRES_DB: ${KONG_PG_DATABASE:-kong}
      POSTGRES_PASSWORD: ${KONG_PG_PASSWORD:-kong}
      POSTGRES_USER: ${KONG_PG_USER:-kong}
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "${KONG_PG_USER:-kong}"]
      interval: 30s
      timeout: 30s
      retries: 3
    restart: on-failure
    stdin_open: true
    tty: true
    networks:
      - kong-net
    volumes:
      - kong_data:/var/lib/postgresql/data
  konga:
    image: pantsel/konga:latest
    container_name: konga
    depends_on:
      - kong
    ports:
      - 1337:1337
    links:
      - mongo:db
      - kong:kong
    environment:
      - DB_ADAPTER=mongo
      - DB_HOST=mongo
      - DB_DATABASE=konga
      - NODE_ENV=development
  mongo:
    image: mongo
    volumes:
      - mongo_data:/data/db
    ports:
      - 127.0.0.1:27017:27017

```

**Comment by pantsel** [comment]
@nohorbee , try to add the `kong-net` network to Konga's config block as well.
```
networks:
      - kong-net
```

**Comment by nohorbee** [comment]
Thanks @pantsel ! I have it running now thanks to your comments + a couple of things that I would like to comment just in case someone (as lacking of experience as me) reads this.
When adding konga to the kong-net, mongo needs to be added too (otherwise Konga won't see it and it won't start). 
After that, I have been trying with localhost, since . . . (now I'm ashamed of that. I can't explain why I didn't realize about localhost not meaning anything for Konga). So, I figured out the IP address of the Kong container. HINT: It's not "ifconfig" inside the container. I got it by running `docker inspect <id-of-docker-container>` and checking the `NetworkSettings.Networks['kong-latest_kong-net'].IPAddress` (of the returned JSON). Using that IP I was able to connect!

Now, it's time to explore and enjoy Konga!
